### PR TITLE
Document that `confirmation_height` is unset

### LINKED ISSUE #1217: confirmation_height is always zero
### Background

while a channel is in pending state lncli pendingchannels always shows "confirmation_height" : 0

### Your environment

lnd 0.41beta
 4.15.0-20-genericx86_64 GNU/Linux
bitcoind 0.16

### Steps to reproduce

open a channel to LND from other client
list channels with lncli pendingchannels
wait for some confirmations

### Expected behaviour

confirmation_height should show the number of confirmations

### Actual behaviour

confirmation_height always show 0

### PR DESCRIPTION
**Pull request by tvolk131**
Fixes #1217

Since we're not currently setting `confirmation_height` anywhere in the codebase, let's remove it.

### CODE REVIEW

**Comment by ziggie1984** [comment]
I would deprecate it and use maturity_height instead if we want to keep it similar to the information we use then funds are in limbo

**Comment by yyforyongyu** [comment]
We might just delete the field if it's not used?

**Comment by tvolk131** [comment]
I updated this PR to remove the field rather than mark it as unset.

File: lnrpc/lightning.proto [hunk 1]
```diff
@@ -2390,6 +2387,8 @@ message PendingChannelsResponse {
         transaction. This value can later be updated once the channel is open.
         */
         int64 fee_per_kw = 6;
+
+        reserved 2;
```
Comment (guggero) [inline lnrpc/lightning.proto:2391]: Can you add a comment to mention what this previously was and to not re-use this argument number please?

**Review by guggero — APPROVED** [review]
LGTM, just one comment about adding a comment.

File: lnrpc/lightning.proto (hunk 1 above)
Comment (yyforyongyu) [reply lnrpc/lightning.proto:2391]: +1

**Review by yyforyongyu — APPROVED** [review]
LGTM👍 Thanks for the fix!

**Comment by tvolk131** [comment]
This is ready to be merged!

**Pull request merged**